Notebook 1 — EDA e Baseline

Objetivo exploratório inicial

Investigar a composição e a qualidade da carteira de crédito no estado do Rio Grande do Sul, utilizando dados agregados do SCR, com foco em entender a distribuição do volume de crédito e seus principais indicadores de risco.

Inicialmente, realiza-se uma análise exploratória estática, considerando um único mês como fotografia da carteira de crédito, com o objetivo de compreender a estrutura e a consistência dos dados. Em etapa posterior, a análise é estendida para a dimensão temporal.

In [28]:
import pandas as pd
import matplotlib.pyplot as plt

In [29]:
df = pd.read_csv('scrdata_202501.csv', sep=";")
df.head()

,data_base,uf,segmento,cliente,cnae_ocupacao,porte,modalidade,submodalidade,origem,indexador,...,a_vencer_de_1081_ate_1800_dias,a_vencer_de_1801_ate_5400_dias,a_vencer_acima_de_5400_dias,carteira_a_vencer,vencido_de_15_ate_90_dias,vencido_acima_de_90_dias,carteira_vencida,carteira_ativa,carteira_inadimplencia,ativo_problematico
0,2025-01-31,AC,Arrendamento,PJ,Comércio; reparação de veículos automotores e ...,Médio,Operações de arrendamento,Arrendamento financeiro exceto veículos automo...,Sem destinação específica,Prefixado,...,"0,00","0,00","0,00","1683314,63","0,00","0,00","0,00","1683314,63","0,00","0,00"
1,2025-01-31,AC,Arrendamento,PJ,Construção,Médio,Operações de arrendamento,Arrendamento financeiro exceto veículos automo...,Sem destinação específica,Prefixado,...,"912187,12","0,00","0,00","3058631,63","0,00","0,00","0,00","3058631,63","0,00","0,00"
2,2025-01-31,AC,Banco,PF,Aposentado/pensionista,Acima de 20 salários mínimos,Empréstimos,"Cartão de crédito - compra, fatura parcelada o...",Sem destinação específica,Prefixado,...,"1486622,67","1146307,44","0,00","4507248,55","0,00","1067,39","1067,39","4508315,94","1067,39","20708,14"
3,2025-01-31,AC,Banco,PF,Aposentado/pensionista,Acima de 20 salários mínimos,Empréstimos,Cartão de crédito - não migrado,Sem destinação específica,Prefixado,...,"11477,91","5806,11","0,00","41068,29","14075,27","93696,32","107771,59","148839,88","93696,32","122950,53"
4,2025-01-31,AC,Banco,PF,Aposentado/pensionista,Acima de 20 salários mínimos,Empréstimos,Cheque especial,Sem destinação específica,Prefixado,...,"0,00","0,00","0,00","211924,89","0,00","15787,88","15787,88","227712,77","15787,88","16239,96"


Antes de iniciar qualquer exploração é preciso estudar o df para se familiarizar com ele.
Este df trás dados agregados de operações de crédito.
Ele não apresenta dados de indivíduos ou empresas, mas sim, várias variáveis, referente a crédido, que se agrupadas revelam informações de um segmento, ou uma modalidade de operação ou de uma região, etc.
Importante sempre acessar o dicionário que explica o df e cada uma de suas variáveis.
Complementar a isso, eu realizei a leitura do artigo: xxxxxx para me aprofundar na temática de inadimplencia.

In [30]:
df_rs = df.loc[df['uf'] == 'RS'].copy()

In [31]:
df_rs[['carteira_ativa', 'carteira_a_vencer', 'carteira_vencida']].head(10)

,carteira_ativa,carteira_a_vencer,carteira_vencida
229040,"117605,30","117605,30","0,00"
229041,"8477993,35","8477993,35","0,00"
229042,"239418,80","239418,80","0,00"
229043,"3870213,75","3870213,75","0,00"
229044,"91459,87","91459,87","0,00"
229045,"2409880,18","2409880,18","0,00"
229046,"37499,22","37499,22","0,00"
229047,"582435,68","582435,68","0,00"
229048,"6111254,27","6111254,27","0,00"
229049,"12929733,82","12929733,82","0,00"


Todos os valores serão convertidos para o sistema internacional, que não usa vírguas e sim, pontos.

In [32]:
df_rs['carteira_ativa'] = df_rs['carteira_ativa'].astype(str)


In [33]:
df_rs['carteira_ativa'] = (
    df_rs['carteira_ativa']
        .str.replace('.', '', regex=False)
        .str.replace(',', '.', regex=False)
)



In [34]:
df_rs['carteira_ativa'] = df_rs['carteira_ativa'].astype(float)

In [35]:
df_rs.head()

,data_base,uf,segmento,cliente,cnae_ocupacao,porte,modalidade,submodalidade,origem,indexador,...,a_vencer_de_1081_ate_1800_dias,a_vencer_de_1801_ate_5400_dias,a_vencer_acima_de_5400_dias,carteira_a_vencer,vencido_de_15_ate_90_dias,vencido_acima_de_90_dias,carteira_vencida,carteira_ativa,carteira_inadimplencia,ativo_problematico
229040,2025-01-31,RS,Arrendamento,PF,Aposentado/pensionista,Acima de 20 salários mínimos,Operações de arrendamento,Arrendamento financeiro exceto veículos automo...,Sem destinação específica,Flutuantes,...,"0,00","0,00","0,00","117605,30","0,00","0,00","0,00",117605.30,"0,00","0,00"
229041,2025-01-31,RS,Arrendamento,PF,Empresário,Acima de 20 salários mínimos,Operações de arrendamento,Arrendamento financeiro exceto veículos automo...,Sem destinação específica,Flutuantes,...,"0,00","0,00","0,00","8477993,35","0,00","0,00","0,00",8477993.35,"0,00","0,00"
229042,2025-01-31,RS,Arrendamento,PJ,"Agricultura, pecuária, produção florestal, pes...",Médio,Operações de arrendamento,Arrendamento financeiro de veículos automotores,Sem destinação específica,Prefixado,...,"0,00","0,00","0,00","239418,80","0,00","0,00","0,00",239418.80,"0,00","0,00"
229043,2025-01-31,RS,Arrendamento,PJ,"Agricultura, pecuária, produção florestal, pes...",Médio,Operações de arrendamento,Arrendamento financeiro exceto veículos automo...,Sem destinação específica,Flutuantes,...,"0,00","0,00","0,00","3870213,75","0,00","0,00","0,00",3870213.75,"0,00","0,00"
229044,2025-01-31,RS,Arrendamento,PJ,"Agricultura, pecuária, produção florestal, pes...",Médio,Operações de arrendamento,Arrendamento financeiro exceto veículos automo...,Sem destinação específica,Prefixado,...,"0,00","0,00","0,00","91459,87","0,00","0,00","0,00",91459.87,"0,00","0,00"


In [36]:
df_rs['vencido_de_15_ate_90_dias'] = (
    df_rs['vencido_de_15_ate_90_dias']
        .str.replace('.', '', regex=False)
        .str.replace(',', '.', regex=False)
)
df_rs['vencido_acima_de_90_dias'] = (
    df_rs['vencido_acima_de_90_dias']
        .str.replace('.', '', regex=False)
        .str.replace(',', '.', regex=False)
)

In [37]:
df_rs['carteira_a_vencer'] = (
    df_rs['carteira_a_vencer']
        .str.replace('.', '', regex=False)
        .str.replace(',', '.', regex=False)
)

In [38]:
df_rs['carteira_vencida'] = (
    df_rs['carteira_vencida']
        .str.replace('.', '', regex=False)
        .str.replace(',', '.', regex=False)
)
df_rs['carteira_inadimplencia'] = (
    df_rs['carteira_inadimplencia']
        .str.replace('.', '', regex=False)
        .str.replace(',', '.', regex=False)
)
df_rs['ativo_problematico'] = (
    df_rs['ativo_problematico']
        .str.replace('.', '', regex=False)
        .str.replace(',', '.', regex=False)
)

df_rs['vencido_de_15_ate_90_dias'] = (
    df_rs['vencido_de_15_ate_90_dias']
        .str.replace('.', '', regex=False)
        .str.replace(',', '.', regex=False)
)
df_rs['vencido_acima_de_90_dias'] = (
    df_rs['vencido_acima_de_90_dias']
        .str.replace('.', '', regex=False)
        .str.replace(',', '.', regex=False)
)

In [39]:
df_rs['carteira_a_vencer'] = df_rs['carteira_a_vencer'].astype(float)
df_rs['carteira_vencida'] = df_rs['carteira_vencida'].astype(float)
df_rs['carteira_inadimplencia'] = df_rs['carteira_inadimplencia'].astype(float)
df_rs['ativo_problematico'] = df_rs['ativo_problematico'].astype(float)
df_rs['numero_de_operacoes'] = df_rs['numero_de_operacoes'].astype(float)
df_rs['vencido_de_15_ate_90_dias'] = df_rs['vencido_de_15_ate_90_dias'].astype(float)
df_rs['vencido_acima_de_90_dias'] = df_rs['vencido_acima_de_90_dias'].astype(float)

In [40]:
df_rs[['carteira_ativa', 'carteira_a_vencer', 'carteira_vencida']].tail(10)

,carteira_ativa,carteira_a_vencer,carteira_vencida
247239,216990.22,216990.22,0.00
247240,29036.83,29036.83,0.00
247241,222224.33,222224.33,0.00
247242,159565.20,159565.20,0.00
247243,206061.59,206061.59,0.00
247244,30957619.80,30957619.80,0.00
247245,1978.61,1978.61,0.00
247246,861573.04,695051.48,166521.56
247247,100505.15,0.00,100505.15
247248,5917.01,0.00,5917.01


In [41]:
df_rs.info()

<class 'pandas.core.frame.DataFrame'>
Index: 18209 entries, 229040 to 247248
Data columns (total 24 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   data_base                       18209 non-null  object 
 1   uf                              18209 non-null  object 
 2   segmento                        18209 non-null  object 
 3   cliente                         18209 non-null  object 
 4   cnae_ocupacao                   18209 non-null  object 
 5   porte                           18209 non-null  object 
 6   modalidade                      18209 non-null  object 
 7   submodalidade                   18209 non-null  object 
 8   origem                          18209 non-null  object 
 9   indexador                       18209 non-null  object 
 10  numero_de_operacoes             18209 non-null  float64
 11  a_vencer_ate_90_dias            18209 non-null  object 
 12  a_vencer_de_91_ate_360_dias    

In [42]:
(df_rs['carteira_vencida'] / df_rs['carteira_ativa']).describe()


count    18209.000000
mean         0.136145
std          0.290368
min          0.000000
25%          0.000000
50%          0.000858
75%          0.066893
max          1.000000
dtype: float64

In [43]:
df_rs[['carteira_ativa', 'carteira_a_vencer', 'carteira_vencida']].tail(10)

,carteira_ativa,carteira_a_vencer,carteira_vencida
247239,216990.22,216990.22,0.00
247240,29036.83,29036.83,0.00
247241,222224.33,222224.33,0.00
247242,159565.20,159565.20,0.00
247243,206061.59,206061.59,0.00
247244,30957619.80,30957619.80,0.00
247245,1978.61,1978.61,0.00
247246,861573.04,695051.48,166521.56
247247,100505.15,0.00,100505.15
247248,5917.01,0.00,5917.01


Com os dados consistêntes, realizamos a agregação dos dados, que é a soma das principais métricas.
Por que essas e não outras variáveis? Por que essas são compostas pelo valor de outras variáveis.
A agregação possíbilita vermos total da carteira de crédito do RS, no mês escolhido.

In [44]:
metricas = [
    'carteira_ativa',
    'carteira_a_vencer',
    'carteira_vencida',
    'carteira_inadimplencia',
    'ativo_problematico'
]

rs_agregado = df_rs[metricas].sum()
rs_agregado


carteira_ativa            4.749059e+11
carteira_a_vencer         4.667138e+11
carteira_vencida          8.192060e+09
carteira_inadimplencia    1.114913e+10
ativo_problematico        2.749278e+10
dtype: float64

Inicialmente, a análise é conduzida no nível agregado do estado, com o objetivo de estabelecer um baseline da carteira de crédito. Esse referencial permite, em etapas posteriores, avaliar desvios e assimetrias ao segmentar a carteira por perfil do tomador ou tipo de operação.

In [54]:
pct_carteira_vencida = rs_agregado["carteira_vencida"] / rs_agregado["carteira_ativa"]
pct_carteira_vencida_RS = round(float(pct_carteira_vencida * 100), 2)
pct_carteira_vencida_RS

1.72

O percentual de carteira vencida é um indicador inicial de deterioração da carteira, refletindo atrasos no pagamento. Isoladamente, não caracteriza inadimplência, mas funciona como sinal de alerta quando analisado em conjunto com métricas de inadimplência e ativos problemáticos.
Como vimos o 1.72% da carteira ativa está vencida, um risco baixo para carteira.

In [56]:
pct_carteira_inadimplente = rs_agregado["carteira_inadimplencia"] / rs_agregado["carteira_ativa"]
pct_carteira_inadimplente_RS = round(float(pct_carteira_inadimplente * 100), 2)
pct_carteira_inadimplente_RS

2.35

A carteira ativa do RS tem um percentual de inadimplência de 2.35%. Ou seja, a cada R$100 de crédito, R$2,35 não são pagos. Nesse primeiro momento não temos como dizer se esse valor é alto ou não, ou se está dentro do esperado, pois não temos um valor base para fazer essa comparação. O que ele mostra até aqui é que há inadimplência que não pode ser deixada de lado.

In [57]:
pct_carteira_ativo_problemático = rs_agregado["ativo_problematico"] / rs_agregado["carteira_ativa"]
pct_carteira_ativo_problemático_rs = round(float(pct_carteira_ativo_problemático * 100), 2)
pct_carteira_ativo_problemático_rs

5.79

A carteira ativa possui 5.79% de ativos problemáticos para o mês. Que são aqueles que possúem algum atraso acima de 90 dias ou que passaram por alguma deteriorização da qualidade de crédito.
Ao olhar isoladamente para eles, apresentam um risco não concretizado, é mais uma ameaça a carteira ativa. Quando olharmos em comparação com outros meses será possível estabeler uma relação dele com outros indicadores.

#Conclusão

Até aqui tiramos uma foto da carteira de crédito do RS, o que nos ajudou a entender os dados e identificar os indicadores relevantes para analisarmos a carteira. O próximo passo é investigar a dinâmica temporal entre ativos problemáticos e inadimplência, que será desenvolvida em notebook separado.